xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/XWiki/DeletedDocuments.xml

```diff
@@ -349,7 +390,32 @@ document.observe('xwiki:dom:loaded', function() {
   if ($('documentsTrash')) {
     $('documentsTrash').up('.hidden').removeClassName('hidden');
   }
-});</code>
+});
+
+// we need to wait the complete display of the livetable to get the number of rows
+document.observe('xwiki:livetable:displayComplete', function() {
+  if ($('documentsTrash')) {
+    var actionAllContainer = $('documentsTrash').getElementsBySelector('td.xwiki-livetable-display-header-filter:last-child')[0];
+    var trashAllDisabled = (trashDocumentsLivetable.totalRows > 0) ? false : true;
+
+    // avoid to get multiple times the button
+    if ($('trashAllDeleted')) {
+      $('trashAllDeleted').remove();
+    }
+
+    var trashAll = new Element('a', {
+        'id': 'trashAllDeleted',
+        'class':'btn btn-danger',
+        'href':'#',
+        'disabled': trashAllDisabled
+        }).update('Delete all');
+    actionAllContainer.insert(trashAll);
+    if (!trashAllDisabled) {
```

Comment:
It feels a bit strange to add the listener only when the button is enabled. Normally you don't need to check anything if you use an real button because disabled buttons don't trigger the click event. But since you're using a link (which is OK) there's no native way to disable the link so you need some JavaScript. What I normally do it check the link state from the event listener, which you already do above, so this ``IF`` shouldn't be needed.